AmplifyPlugins/Core/AWSPluginsCore/Model/Support/ModelField+GraphQL.swift

```diff
@@ -0,0 +1,47 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+/// Extension that adds GraphQL specific utilities to `ModelField`.
+extension ModelField {
+
+    /// The GraphQL name of the field.
+    var graphQLName: String {
+        if isAssociationOwner, case let .belongsTo(_, targetName) = association {
+            return targetName ?? name.pascalCased() + "Id"
+        }
+        return name
+    }
+}
+
+/// Extension that translate an array of model fields into `SelectionSet` structure.
+extension Array where Element == ModelField {
```

Comment:
We may be using this pattern elsewhere, but I'm having second thoughts about it. It feels odd to have something so local applied as an extension to a standard type like Array, even with the `ModelField` condition. Consider rewriting these as helper methods on `ModelField`, like

```swift
extension ModelField {
    static func selectionSet(for fields: [ModelField]) -> SelectionSet { ... }
    static func selectionSetFields(for fields: [ModelField]) -> [SelectionSetField] { ... }
}
```

Reply:
I ended up rewriting this as an extension on SelectionSet
```
convenience init(fields: [ModelField])
```
it seems to be a good spot since SelectionSets are always first created with model fields